File: Phase01/test/ResultTest.java
package Phase01.test;

import Phase01.src.*;

import java.util.Arrays;
import java.util.HashSet;

import static org.junit.Assert.assertTrue;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class ResultTest {
    private Result query, word;

    @BeforeEach
    public void createObject() throws Exception {
        Entry[] entryList = { new Entry("test1", 11), new Entry("test2", 14) };
        query = new Result(new HashSet<Entry>(Arrays.asList(entryList)), true);
        word = new Result(new HashSet<Entry>(Arrays.asList(entryList)), false);
    }

    @Test
    public void toStringTest() {
        assertTrue(query.toString().equals(" test1 test2") || query.toString().equals(" test2 test1"));
        assertTrue(word.toString().equals("[Document Id: test1, Index: 11, Document Id: test2, Index: 14]")
                || word.toString().equals("[Document Id: test2, Index: 14, Document Id: test1, Index: 11]"));
    }

}

File: Phase05/app/library.Test/ResultTest.cs
using System.Collections.Generic;
using System;
using Xunit;

namespace library.Test
{
    public class ResultTest
    {
        private Result query, word, empty;

        public ResultTest()
        {
            Entry[] entryList = { new Entry("test1", 11), new Entry("test2", 14) };
            this.query = new Result(new HashSet<Entry>(entryList), true);
            this.word = new Result(new HashSet<Entry>(entryList), false);
            this.empty = new Result(new HashSet<Entry>(), false);
        }

        [Fact]
        public void ToStringTest()
        {
            Assert.True(query.ToString() == " test1 test2" || query.ToString() == " test2 test1");
            Assert.True(word.ToString() == "Document Id: test1, Index: 11, Document Id: test2, Index: 14"
                         || word.ToString() == "[Document Id: test2, Index: 14, Document Id: test1, Index: 11]");
            Assert.True(empty.ToString() == "No result found");
        }
    }
}